### DISCUSSION TITLE
proxmox, lxc, ansible use-case(s)

### DISCUSSION THREAD

**Discussion by jschneekloth** [louislam/dockge#365] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Apologies for this being a bit of a catch-all and not specific to a single feature request as well as the amount of text here....

Also, thanks so much for all the work done to date on uptime and dockge, these are both truly innovative services and love them both!  Let me first describe my use-case and then what I would love to see from Dockge in order to better support that use-case which I also feel would be useful to many other use-cases as well.  My setup is as follows.  I use proxmox as my hypervisor and auto create new LXC's (or VMs) for new services in my home lab.  On each LXC, I install docker and then use docker compose to manage the service(s) that LXC serves.  For example, I have a monitoring LXC, that I use docker (compose) to deploy grafana, prometheus, influxdb, and uptime-kuma for holding metrics and driving alerts for the overall home lab itself.  I use ansible for creating and managing the LXC and docker compose for defining what is on each lxc so I can push it all to source control and rebuild the world at-will if ever needed, not to mention things like tracking changes to services rollbacks and anything else that by such a workflow.  It's worked **really** well so far.  What I have found lacking, however, is a centralized and visual way of tracking all the containers/services across everything in the home lab.  What I have seen from Dockge so far is SO CLOSE to perfection for this use case!!

Here's what I've played with so far:
1. Creating a new LXC from dockge only being deployed to it
2. Configure the services supported on that LXC by publishing my existing docker compose to the stacks directory
3. Create a specific, one-off LXC that is meant to act as the "single" manager for everything else being created

I manually went through this process (via ansible) and created the "dockge-master" lxc that I would expect the agents to roll up to now that I see that's a thing (awesome btw) and I've updated an existing LXC to by dockge first...which means I do the following:
1. manually create container in proxmox
2. provision docker (update debian on lxc, install docker/compose)
3. rely on ansible to push docker compose that sets up dockge, with stack folder specified
4. copy existing docker compose to the stack folder from (3)

And this works!  I can then go to dockge url, create username/password, manually add it to the "single" manager via the agent and manually start the discovered docker compose.  This is a great step forward in my overall experience.  Now....here is how I view this overall UX can be improved, and I suspect this is more globally useful than specific to how I want to use Dockge as well (I think in priority order from my perspective).

1. Allow username/password to be defined via configuration in docker compose
2. Starting the  stacks that exist within the /stacks dir by default, rather than relying upon manual hitting endpoint and clicking start on the stack
3. Allow agents to be discovered via config setup (compose or .env of master), rather than manually adding a new agent
4. Automagical emission of metrics to grafana via influx/etc for container health.

I think with above alone, are more precise and manageable workflow can be supported.

More general quality of life improvements:
1. Prioritize running stack logs....when selecting a container in log, only show those logs in window
2. Prioritize running stack logs....resizing of this section of the window...it is almost the most important day-to-day interaction, should make it front and center from UX perspective
3. Prioritize running stack logs....add timestamps to output
6. Integration with uptime?  I would love to be notified when updates to containers are available, or maybe even automatic updates ala watchtower (would love to ditch watchtower)

Thanks again for everything in this project and uptime!  Happy to walk through my use case in more detail as well...

### ✔️ Solution

no solution explicitly, just starting a convo

**Comment by bs-github** [comment]
Hi @jschneekloth,
interesting use case.
Would you **please share your ansible code** to set up the LXC and dockage stuff?
I'm quite interested in that part.
